[DE-616] ttl for cursor cannot be configured from ArangoDB Spark data source and getting cursor not found

I am writing a data pipeline to ingest data from Arango db to Bigquery. I have used the arango spark data source. 

here is the code :
```
 df: DataFrame = spark.read.format("com.arangodb.spark") \
        .option("query", query) \
        .options(**arango_connection) \
        .schema(doc_schema).load()
df.count()
    df.write.format('bigquery').mode("append") \
        .option('table', bq_table) \
        .option("project", bq_project) \
        .option("dataset", bq_dataset) \
        .option("writeMethod", "direct") \
        .option('credentialsFile', 'path/of/gcp-credential') \
        .save() 
```

this code works if my Arango collection has fewer documents, like writing 10000 documents from Arango db to Bigquery.

But my collection will have more than 10, 00, 000 documents. When testing with above 50, 000 document spark job fails.

error getting 
Caused by: org.apache.spark.util.TaskCompletionListenerException: Response: 404, Error: 1600 - cursor not found


org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 0.0 failed 1 times, most recent failure: Lost task 0.0 in stage 0.0 (TID 0) (BS1ZN93 executor driver): org.apache.spark.util.TaskCompletionListenerException: Response: 404, Error: 1600 - cursor not found
	at org.apache.spark.TaskContextImpl.invokeListeners(TaskContextImpl.scala:254)
	at org.apache.spark.TaskContextImpl.invokeTaskCompletionListeners(TaskContextImpl.scala:144)
	at org.apache.spark.TaskContextImpl.markTaskCompleted(TaskContextImpl.scala:137)
	at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:180)
	at org.apache.spark.scheduler.Task.run(Task.scala:139)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:554)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1529)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:557)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
	Suppressed: com.arangodb.ArangoDBException: Response: 404, Error: 1600 - cursor not found
		at com.arangodb.internal.util.ResponseUtils.checkError(ResponseUtils.java:53)
		at com.arangodb.http.HttpCommunication.execute(HttpCommunication.java:86)
		at com.arangodb.http.HttpCommunication.execute(HttpCommunication.java:66)
		at com.arangodb.http.HttpProtocol.execute(HttpProtocol.java:44)
		at com.arangodb.internal.ArangoExecutorSync.execute(ArangoExecutorSync.java:60)
		at com.arangodb.internal.ArangoExecutorSync.execute(ArangoExecutorSync.java:48)
		at com.arangodb.internal.ArangoDatabaseImpl$1.close(ArangoDatabaseImpl.java:219)
		at com.arangodb.internal.cursor.ArangoCursorImpl.close(ArangoCursorImpl.java:69)
		at org.apache.spark.sql.arangodb.datasource.reader.ArangoQueryReader.close(ArangoQueryReader.scala:53)
		at org.apache.spark.sql.execution.datasources.v2.DataSourceRDD$$anon$1.$anonfun$advanceToNextIter$1(DataSourceRDD.scala:94)
		at org.apache.spark.sql.execution.datasources.v2.DataSourceRDD$$anon$1.$anonfun$advanceToNextIter$1$adapted(DataSourceRDD.scala:89)
		at org.apache.spark.TaskContext$$anon$1.onTaskCompletion(TaskContext.scala:132)
		at org.apache.spark.TaskContextImpl.$anonfun$invokeTaskCompletionListeners$1(TaskContextImpl.scala:144)
		at org.apache.spark.TaskContextImpl.$anonfun$invokeTaskCompletionListeners$1$adapted(TaskContextImpl.scala:144)
		at org.apache.spark.TaskContextImpl.invokeListeners(TaskContextImpl.scala:199)
		... 10 more

Driver stacktrace:
	at org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2785)
	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:2721)
	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:2720)
	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:2720)
	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:1206)
	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:1206)
	at scala.Option.foreach(Option.scala:407)
	at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:1206)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:2984)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2923)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2912)
	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)
	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:971)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2263)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2284)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2303)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2328)
	at org.apache.spark.rdd.RDD.$anonfun$collect$1(RDD.scala:1019)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
	at org.apache.spark.rdd.RDD.withScope(RDD.scala:405)
	at org.apache.spark.rdd.RDD.collect(RDD.scala:1018)
	at org.apache.spark.sql.execution.SparkPlan.executeCollect(SparkPlan.scala:448)
	at org.apache.spark.sql.execution.adaptive.AdaptiveSparkPlanExec.$anonfun$executeCollect$1(AdaptiveSparkPlanExec.scala:354)
	at org.apache.spark.sql.execution.adaptive.AdaptiveSparkPlanExec.withFinalPlanUpdate(AdaptiveSparkPlanExec.scala:382)
	at org.apache.spark.sql.execution.adaptive.AdaptiveSparkPlanExec.executeCollect(AdaptiveSparkPlanExec.scala:354)
	at org.apache.spark.sql.Dataset.$anonfun$count$1(Dataset.scala:3459)
	at org.apache.spark.sql.Dataset.$anonfun$count$1$adapted(Dataset.scala:3458)
	at org.apache.spark.sql.Dataset.$anonfun$withAction$2(Dataset.scala:4167)
	at org.apache.spark.sql.execution.QueryExecution$.withInternalError(QueryExecution.scala:526)
	at org.apache.spark.sql.Dataset.$anonfun$withAction$1(Dataset.scala:4165)
	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$6(SQLExecution.scala:118)
	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:195)
	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:103)
	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:827)
	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:65)
	at org.apache.spark.sql.Dataset.withAction(Dataset.scala:4165)
	at org.apache.spark.sql.Dataset.count(Dataset.scala:3458)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
	at py4j.Gateway.invoke(Gateway.java:282)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
	at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: org.apache.spark.util.TaskCompletionListenerException: Response: 404, Error: 1600 - cursor not found
	at org.apache.spark.TaskContextImpl.invokeListeners(TaskContextImpl.scala:254)
	at org.apache.spark.TaskContextImpl.invokeTaskCompletionListeners(TaskContextImpl.scala:144)
	at org.apache.spark.TaskContextImpl.markTaskCompleted(TaskContextImpl.scala:137)
	at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:180)
	at org.apache.spark.scheduler.Task.run(Task.scala:139)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:554)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1529)
	at org.apac 
	
	There is no option found to configure ttl which is the default 30 sec. I tried to reduce batch size but still getting cursor not found iss

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -45,24 +45,24 @@ jobs:
           - 8
           - 11
         docker-img:
-          - docker.io/arangodb/arangodb:3.10.10
-          - docker.io/arangodb/enterprise:3.10.10
-          - docker.io/arangodb/arangodb:3.11.3
-          - docker.io/arangodb/enterprise:3.11.3
+          - docker.io/arangodb/arangodb:3.10.13
+          - docker.io/arangodb/enterprise:3.10.13
+          - docker.io/arangodb/arangodb:3.11.8
+          - docker.io/arangodb/enterprise:3.11.8
         exclude:
           - scala-version: 2.13
             spark-version: 3.1
-          - docker-img: docker.io/arangodb/arangodb:3.10.10
+          - docker-img: docker.io/arangodb/arangodb:3.10.13
             java-version: 8
-          - docker-img: docker.io/arangodb/enterprise:3.10.10
+          - docker-img: docker.io/arangodb/enterprise:3.10.13
             java-version: 8
-          - docker-img: docker.io/arangodb/enterprise:3.10.10
+          - docker-img: docker.io/arangodb/enterprise:3.10.13
             topology: single
-          - docker-img: docker.io/arangodb/arangodb:3.11.3
+          - docker-img: docker.io/arangodb/arangodb:3.11.8
             java-version: 11
-          - docker-img: docker.io/arangodb/enterprise:3.11.3
+          - docker-img: docker.io/arangodb/enterprise:3.11.8
             java-version: 8
-          - docker-img: docker.io/arangodb/enterprise:3.11.3
+          - docker-img: docker.io/arangodb/enterprise:3.11.8
             topology: single
 
     steps:
@@ -102,7 +102,7 @@ jobs:
         java-version:
           - 8
         docker-img:
-          - docker.io/arangodb/arangodb:3.11.3
+          - docker.io/arangodb/arangodb:3.11.8
         exclude:
           - scala-version: 2.13
             spark-version: 3.1
@@ -138,7 +138,7 @@ jobs:
         spark-version: [3.1, 3.2, 3.3, 3.4]
         topology: [single, cluster]
         java-version: [8, 11]
-        docker-img: ["docker.io/arangodb/arangodb:3.11.3"]
+        docker-img: ["docker.io/arangodb/arangodb:3.11.8"]
         exclude:
           - topology: cluster
             java-version: 8
@@ -197,7 +197,7 @@ jobs:
           - 8
           - 11
         docker-img:
-          - docker.io/arangodb/arangodb:3.11.3
+          - docker.io/arangodb/arangodb:3.11.8
         exclude:
           - scala-version: 2.13
             spark-version: 3.1
@@ -305,7 +305,7 @@ jobs:
         run: ./docker/start_db.sh
         env:
           STARTER_MODE: cluster
-          DOCKER_IMAGE: docker.io/arangodb/arangodb:3.11.3
+          DOCKER_IMAGE: docker.io/arangodb/arangodb:3.11.8
       - name: Info
         run: mvn -version
       - name: Install
@@ -313,48 +313,49 @@ jobs:
       - name: Deployment Test
         run: mvn -f ./demo/pom.xml -Pscala-${{matrix.scala-version}} -Pspark-${{matrix.spark-version}} -Dspark.version=${{matrix.spark-full-version}} -DimportPath=docker/import test
 
-  sonar:
-    timeout-minutes: 10
-    runs-on: ubuntu-latest
-
-    strategy:
-      fail-fast: false
-      matrix:
-        scala-version:
-          - 2.12
-        spark-version:
-          - 3.4
-        topology:
-          - single
-        java-version:
-          - 11
-        docker-img:
-          - docker.io/arangodb/arangodb:3.11.3
-
-    steps:
-      - uses: actions/checkout@v2
-        with:
-          fetch-depth: 0
-      - uses: actions/setup-java@v2
-        with:
-          java-version: ${{matrix.java-version}}
-          distribution: 'adopt'
-          cache: maven
-      - name: Start Database
-        run: ./docker/start_db.sh
-        env:
-          STARTER_MODE: ${{matrix.topology}}
-          DOCKER_IMAGE: ${{matrix.docker-img}}
-      - name: Cache SonarCloud packages
-        uses: actions/cache@v2
-        with:
-          path: ~/.sonar/cache
-          key: ${{ runner.os }}-sonar
-          restore-keys: ${{ runner.os }}-sonar
-      - name: Info
-        run: mvn -version
-      - name: Build and analyze
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
-          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
-        run: mvn -e --no-transfer-progress -Pscala-${{matrix.scala-version}} -Pspark-${{matrix.spark-version}} -Dgpg.skip=true -B verify org.sonarsource.scanner.maven:sonar-maven-plugin:sonar -Dsonar.projectKey=arangodb_arangodb-spark-datasource
+## FIXME: sonar plugin requires Java 17
+#  sonar:
+#    timeout-minutes: 10
+#    runs-on: ubuntu-latest
+#
+#    strategy:
+#      fail-fast: false
+#      matrix:
+#        scala-version:
+#          - 2.12
+#        spark-version:
+#          - 3.4
+#        topology:
+#          - single
+#        java-version:
+#          - 11
+#        docker-img:
+#          - docker.io/arangodb/arangodb:3.11.8
+#
+#    steps:
+#      - uses: actions/checkout@v2
+#        with:
+#          fetch-depth: 0
+#      - uses: actions/setup-java@v2
+#        with:
+#          java-version: ${{matrix.java-version}}
+#          distribution: 'adopt'
+#          cache: maven
+#      - name: Start Database
+#        run: ./docker/start_db.sh
+#        env:
+#          STARTER_MODE: ${{matrix.topology}}
+#          DOCKER_IMAGE: ${{matrix.docker-img}}
+#      - name: Cache SonarCloud packages
+#        uses: actions/cache@v2
+#        with:
+#          path: ~/.sonar/cache
+#          key: ${{ runner.os }}-sonar
+#          restore-keys: ${{ runner.os }}-sonar
+#      - name: Info
+#        run: mvn -version
+#      - name: Build and analyze
+#        env:
+#          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
+#          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+#        run: mvn -e --no-transfer-progress -Pscala-${{matrix.scala-version}} -Pspark-${{matrix.spark-version}} -Dgpg.skip=true -B verify org.sonarsource.scanner.maven:sonar-maven-plugin:sonar -Dsonar.projectKey=arangodb_arangodb-spark-datasource

--- a/arangodb-spark-commons/src/main/scala/org/apache/spark/sql/arangodb/commons/ArangoClient.scala
+++ b/arangodb-spark-commons/src/main/scala/org/apache/spark/sql/arangodb/commons/ArangoClient.scala
@@ -21,6 +21,7 @@ class ArangoClient(options: ArangoDBConf) extends Logging {
   private def aqlOptions(): AqlQueryOptions = {
     val opt = new AqlQueryOptions()
       .stream(options.readOptions.stream)
+      .ttl(options.readOptions.ttl)
       .fillBlockCache(options.readOptions.fillBlockCache)
       .batchSize(options.readOptions.batchSize)
     opt

--- a/arangodb-spark-commons/src/main/scala/org/apache/spark/sql/arangodb/commons/ArangoDBConf.scala
+++ b/arangodb-spark-commons/src/main/scala/org/apache/spark/sql/arangodb/commons/ArangoDBConf.scala
@@ -155,6 +155,13 @@ object ArangoDBConf {
     .booleanConf
     .createWithDefault(true)
 
+  val TTL = "ttl"
+  val DEFAULT_TTL = 30
+  val ttlConf: ConfigEntry[Int] = ConfigBuilder(TTL)
+    .doc("cursor ttl in seconds")
+    .intConf
+    .createWithDefault(DEFAULT_TTL)
+
   val PARSE_MODE = "mode"
   val parseModeConf: ConfigEntry[String] = ConfigBuilder(PARSE_MODE)
     .doc("allows a mode for dealing with corrupt records during parsing")
@@ -273,6 +280,7 @@ object ArangoDBConf {
     SAMPLE_SIZE -> sampleSizeConf,
     FILL_BLOCK_CACHE -> fillBlockCacheConf,
     STREAM -> streamConf,
+    TTL -> ttlConf,
     PARSE_MODE -> parseModeConf,
     COLUMN_NAME_OF_CORRUPT_RECORD -> columnNameOfCorruptRecordConf,
 
@@ -530,6 +538,8 @@ class ArangoDBReadConf(opts: Map[String, String]) extends ArangoDBConf(opts) {
   val fillBlockCache: Boolean = getConf(fillBlockCacheConf)
 
   val stream: Boolean = getConf(streamConf)
+
+  val ttl: Int = getConf(ttlConf)
 
   val parseMode: ParseMode = ParseMode.fromString(getConf(parseModeConf))
 

--- a/integration-tests/src/test/scala/org/apache/spark/sql/arangodb/datasource/ReadTest.scala
+++ b/integration-tests/src/test/scala/org/apache/spark/sql/arangodb/datasource/ReadTest.scala
@@ -229,9 +229,9 @@ class ReadTest extends BaseSparkTest {
   def readTtl(protocol: String, contentType: String): Unit = {
     val query =
       """
-        |FOR i IN 1..3
-        |RETURN { value: SLEEP(2) }
-        |""".stripMargin.replaceAll("\n", "")
+        |FOR i IN 1..10
+        |RETURN { value: null }
+        |""".stripMargin.replaceAll("\n", " ")
 
     val df = spark.read
       .format(BaseSparkTest.arangoDatasource)
@@ -248,18 +248,11 @@ class ReadTest extends BaseSparkTest {
       .load()
 
     val thrown: Throwable = catchThrowable(new ThrowingCallable() {
-      override def call(): Unit = df.show()
+      override def call(): Unit = df.foreach(_ => Thread.sleep(2000L))
     })
 
-    assertThat(thrown)
-      .isInstanceOf(classOf[SparkException])
-
-    assertThat(thrown.getCause.getSuppressed).isNotEmpty
-    assertThat(thrown.getCause.getSuppressed.head).isInstanceOf(classOf[ArangoDBException])
-
-    val aEx = thrown.getCause.getSuppressed.head.asInstanceOf[ArangoDBException]
-    assertThat(aEx.getResponseCode).isEqualTo(404)
-    assertThat(aEx.getErrorNum).isEqualTo(1600)
-    assertThat(aEx.getMessage).contains("cursor not found")
+    assertThat(thrown).isInstanceOf(classOf[SparkException])
+    assertThat(thrown.getMessage).contains("Error: 1600")
+    assertThat(thrown.getMessage).contains("cursor not found")
   }
 }

--- a/integration-tests/src/test/scala/org/apache/spark/sql/arangodb/datasource/ReadTest.scala
+++ b/integration-tests/src/test/scala/org/apache/spark/sql/arangodb/datasource/ReadTest.scala
@@ -49,12 +49,12 @@ class ReadTest extends BaseSparkTest {
   @Test
   def readCollectionSql(): Unit = {
     val litalien = spark.sql(
-      """
-        |SELECT likes
-        |FROM users
-        |WHERE name == ("Prudence" AS first, "Litalien" AS last)
-        |    AND likes == ARRAY("swimming", "chess")
-        |""".stripMargin)
+        """
+          |SELECT likes
+          |FROM users
+          |WHERE name == ("Prudence" AS first, "Litalien" AS last)
+          |    AND likes == ARRAY("swimming", "chess")
+          |""".stripMargin)
       .first()
     assertThat(litalien.get(0)).isEqualTo(Seq("swimming", "chess"))
   }
@@ -139,7 +139,7 @@ class ReadTest extends BaseSparkTest {
     val badRecords = df.filter("badRecord IS NOT NULL").persist()
       .select("badRecord")
       .collect()
-      .map(_ (0).asInstanceOf[String])
+      .map(_(0).asInstanceOf[String])
 
     assertThat(badRecords).hasSize(1)
     assertThat(badRecords.head).contains(""""v":"3""")
@@ -191,7 +191,7 @@ class ReadTest extends BaseSparkTest {
 
   @ParameterizedTest
   @MethodSource(Array("provideProtocolAndContentType"))
-  def reatTimeout(protocol: String, contentType: String): Unit = {
+  def readTimeout(protocol: String, contentType: String): Unit = {
     val query =
       """
         |RETURN { value: SLEEP(5) }
@@ -224,4 +224,42 @@ class ReadTest extends BaseSparkTest {
       .isInstanceOf(classOf[TimeoutException])
   }
 
+  @ParameterizedTest
+  @MethodSource(Array("provideProtocolAndContentType"))
+  def readTtl(protocol: String, contentType: String): Unit = {
+    val query =
+      """
+        |FOR i IN 1..3
+        |RETURN { value: SLEEP(2) }
+        |""".stripMargin.replaceAll("\n", "")
+
+    val df = spark.read
+      .format(BaseSparkTest.arangoDatasource)
+      .schema(StructType(Array(
+        StructField("value", NullType)
+      )))
+      .options(options + (
+        ArangoDBConf.QUERY -> query,
+        ArangoDBConf.PROTOCOL -> protocol,
+        ArangoDBConf.CONTENT_TYPE -> contentType,
+        ArangoDBConf.BATCH_SIZE -> "1",
+        ArangoDBConf.TTL -> "1"
+      ))
+      .load()
+
+    val thrown: Throwable = catchThrowable(new ThrowingCallable() {
+      override def call(): Unit = df.show()
+    })
+
+    assertThat(thrown)
+      .isInstanceOf(classOf[SparkException])
+
+    assertThat(thrown.getCause.getSuppressed).isNotEmpty
+    assertThat(thrown.getCause.getSuppressed.head).isInstanceOf(classOf[ArangoDBException])
+
+    val aEx = thrown.getCause.getSuppressed.head.asInstanceOf[ArangoDBException]
+    assertThat(aEx.getResponseCode).isEqualTo(404)
+    assertThat(aEx.getErrorNum).isEqualTo(1600)
+    assertThat(aEx.getMessage).contains("cursor not found")
+  }
 }

